src/annotator/adder.tsx

```diff
@@ -1,38 +1,38 @@
 import { render } from 'preact';
 
 import AdderToolbar from './components/AdderToolbar';
+import type { Command } from './components/AdderToolbar';
 import { isTouchDevice } from '../shared/user-agent';
+import type { Destroyable } from '../types/annotator';
+
 import { createShadowRoot } from './util/shadow-root';
 
 /**
- *  @typedef {1} ArrowPointingDown
  * Show the adder above the selection with an arrow pointing down at the
  * selected text.
  */
 export const ARROW_POINTING_DOWN = 1;
 
 /**
- *  @typedef {2} ArrowPointingUp
  * Show the adder above the selection with an arrow pointing up at the
  * selected text.
  */
 export const ARROW_POINTING_UP = 2;
 
-/**
- *  @typedef {ArrowPointingDown|ArrowPointingUp} ArrowDirection
- * Show the adder above the selection with an arrow pointing up at the
- * selected text.
- */
+type ArrowPointingDown = typeof ARROW_POINTING_DOWN;
+type ArrowPointingUp = typeof ARROW_POINTING_UP;
+type ArrowDirection = ArrowPointingUp | ArrowPointingDown;
 
-/**
- * @typedef Target
- * @prop {number} left - Offset from left edge of viewport.
- * @prop {number} top - Offset from top edge of viewport.
- * @prop {ArrowDirection} arrowDirection - Direction of the adder's arrow.
- */
+type Target = {
+  /** Offset from left edge of viewport */
+  left: number;
+  /** Offset from top edge of viewport */
+  top: number;
+  /** Direction of the adder's arrow */
+  arrowDirection: ArrowDirection;
+};
 
-/** @param {number} pixels */
-function toPx(pixels) {
+function toPx(pixels: number) {
```

Comment:
Outside of the scope of a pure conversion, but this utility function is of limited value. It's only used by one method, and in these days of template literals, I think just `${left - parentRect.left}px` is less overhead than an abstracted function here. But obviously not related to a TypeScript conversion :)